CONTRIBUTING.md

```diff
@@ -1,61 +1,129 @@
+* [Run `oad` locally from sources](#run-oad-locally-from-sources)
+* [Install `oad` globally from sources or npm feed](#install-oad-globally-from-sources-or-npm-feed)
+  * [Uninstall `oad` globally](#uninstall-oad-globally)
+* [Purge the obsolete `oad` package from your system](#purge-the-obsolete-oad-package-from-your-system)
+
+<!-- <small><i><a href='http://ecotrust-canada.github.io/markdown-toc/'>
+Table of contents generated with markdown-toc</a></i></small> -->
 
 ## Prerequisites
 
-- [Node.js](https://nodejs.org/) (14.x or higher)
-- [.NET runtime and SDK](https://aka.ms/dotnet-download)
-- [.NET CLI tools](https://github.com/dotnet/cli/releases) version 2.0.0 or higher
+To execute any instructions in this file, first ensure you fulfill all the following prerequisites:
 
-## Build scripts
+1. Install [Node.js](https://nodejs.org/), version 14.x or higher.
+1. Install [.NET runtime and SDK](https://aka.ms/dotnet-download), version 6 or higher.
+1. Install [.NET CLI tools](https://github.com/dotnet/cli/releases) version 2.0.0 or higher.
+1. Execute all commands in this file from your [`openapi-diff`] git repo local clone root dir.
+1. Run `npm install` to install the required node modules.
 
-Run `npm install` to install the required modules.
+## Build the code
 
-```sh
-npm install
-```
+`openapi-diff` is composed both of TypeScript and .NET/C# projects, all of which have to be built
+to be able to run the tool locally from sources.
 
-## How to build the C# code
+The core logic for `openapi-diff` is written in C# and compiled to a .NET binary.  
+The CLI for `openapi-diff` is written in TypeScript and compiled to a Node.js binary.
 
-The core logic for openapi-diff is written in C# and compiled to a .NET binary.
+| Build command | Effect |
+|-|-|
+| `npm run dn.build` | Runs [`dotnet build`] and related commands to build the C# code. |
+| `npm run tsc` | Builds the TypeScript code. |
 
-Use `dn.build` npm script to build the C# code.
+For more commands, see the [`package.json`](/package.json) `scripts` element.
+Notably:
 
-```sh
-npm run dn.build
-```
+| Command | Effect |
+|-|-|
+| `npm run dn.restore` | Runs [`dotnet restore`] in preparation to build the C# code. |
+| `npm run dn.test` | Runs [`dotnet test`] and related commands to test the C# code. |
 
-## How to test the C# code
+### Troubleshoot
 
-To run all tests under the repo
+If the `npm run dn.build` command outputs error like:
 
-```sh
-npm run dn.test
-```
+> C:\Program Files\dotnet\sdk\8.0.100\Sdks\Microsoft.NET.Sdk\targets\Microsoft.PackageDependencyResolution.targe
+       ts(266,5): error NETSDK1005: Assets file 'C:\(...)\openapi-diff\src\core\OpenApiDiff\ob
+       j\project.assets.json' doesn't have a target for 'netcoreapp6.0'. Ensure that restore has run and that you have
+       included 'netcoreapp6.0' in the TargetFrameworks for your project.
+
+Then first run `npm run dn.restore` and then run the `npm run dn.build` command again.
```

Comment:
Are we really wrapping the dotnet restore command in npm 🤦

Reply:
Yeah, all these `dn` in `npm run` stand for `dotnet`. Let's fix it in a future PR :)